src/stdlib/decode_charset.rs

```diff
@@ -0,0 +1,153 @@
+use crate::diagnostic::Label;
+use crate::prelude::*;
+use encoding_rs::Encoding;
+use nom::AsBytes;
+use std::str::from_utf8;
+
+#[derive(Clone, Copy, Debug)]
+pub struct DecodeCharset;
+
+impl Function for DecodeCharset {
+    fn identifier(&self) -> &'static str {
+        "decode_charset"
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "Decode charset from euc-kr",
+                source: r#"decode_charset!(decode_base64!("vsiz58fPvLy/5A=="), "euc-kr")"#,
+                result: Ok("안녕하세요"),
+            },
+            Example {
+                title: "Decode charset from euc-jp",
+                source: r#"decode_charset!(decode_base64!("pLOk86TLpMGkzw=="), "euc-jp")"#,
+                result: Ok("こんにちは"),
+            },
+            Example {
+                title: "Decode charset from gb2312",
+                source: r#"decode_charset!(decode_base64!("xOO6ww=="), "gb2312")"#,
+                result: Ok("你好"),
+            },
+        ]
+    }
+
+    fn summary(&self) -> &'static str {
+        "Decode non UTF-8 charset to UTF-8"
+    }
+
+    fn usage(&self) -> &'static str {
+        indoc! {"
+            Decode non UTF-8 charset to UTF-8.
+
+            The `value` parameter is a non UTF-8 encoded string.
+            The `from_charset` parameter specifies the charset of the `value`.
+        "}
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "from_charset",
+                kind: kind::BYTES,
+                required: true,
+            },
+        ]
+    }
+
+    fn compile(
+        &self,
+        _state: &TypeState,
+        _ctx: &mut FunctionCompileContext,
+        arguments: ArgumentList,
+    ) -> Compiled {
+        let value = arguments.required("value");
+        let from_charset = arguments.required("from_charset");
+
+        Ok(DecodeCharsetFn {
+            value,
+            from_charset,
+        }
+        .as_expr())
+    }
+}
+
+fn decode_charset(value: &[u8], from_charset: &[u8]) -> Resolved {
+    let decoder = Encoding::for_label(from_charset).ok_or_else(|| error(from_charset))?;
+
+    let (output, _, _) = decoder.decode(value);
+    Ok(Value::Bytes(output.as_bytes().to_vec().into()))
+}
+
+fn error(from: &[u8]) -> ExpressionError {
```

Comment:
nit: 
```suggestion
fn create_error(from_charset: &[u8]) -> ExpressionError {
```

Reply:
hank you for your interest. 
I have modified it in the commit. d4be5d7a5712e4bf6c89104c35795f8f0e719cfa